module/VuFind/src/VuFind/Search/Solr/Results.php

```diff
@@ -346,9 +349,21 @@ public function getFacetList($filter = null)
                     $displayText = $this->hierarchicalFacetHelper
                         ->formatDisplayText($displayText);
                 }
-                $currentSettings['displayText'] = $translate
-                    ? $this->translate([$translateTextDomain, $displayText])
-                    : $displayText;
+                $currentSettings['displayText'] = $displayText;
+                if ($translate) {
+                    $translated = $this->translate(
+                        [$translateTextDomain, $displayText]
+                    );
+                    // Apply a format to the translation (if available)
+                    if ($translateFormat) {
+                        $translated = str_replace(
```

Comment:
What if, instead of doing a `str_replace` here, we use this instead to take advantage of our standard translation token system?

```
$translated = $this->translate(
    $translateFormat,
    ['%%raw%%' => $displayText, '%%translated%%" => $translated]
);
```

Now, the format descriptor can be a language file token which incorporates %%raw%% and/or %%translated%%, and this enables us to use different formats for different languages if necessary. I'm pretty sure that including the full raw format string in the .ini file instead of putting it in the language files would also still work, though I'd want to double check that to be sure I'm not forgetting some important detail.

(Note that I switched to double-percent-signs on the tokens for consistency with other tokenized language strings).

Reply:
Done!